examples/image_segmentation/ios/ImageSegmenter/ViewContoller/MediaLibraryViewController.swift

```diff
@@ -0,0 +1,384 @@
+  }
+
+  private func openMediaLibrary() {
+    configurePickerController()
+    present(pickerController, animated: true)
+  }
+
+  private func clearPlayerView() {
+    imageEmptyLabel.isHidden = false
+    removePlayerViewController()
+  }
+
+  private func showProgressView() {
+    guard let progressSuperview = progressView.superview?.superview else {
+      return
+    }
+    progressSuperview.isHidden = false
+    progressView.progress = 0.0
+    progressView.observedProgress = nil
+    self.view.bringSubviewToFront(progressSuperview)
+  }
+
+  private func hideProgressView() {
+    guard let progressSuperview = progressView.superview?.superview else {
+      return
+    }
+    self.view.sendSubviewToBack(progressSuperview)
+    self.progressView.superview?.superview?.isHidden = true
+  }
+
+  func layoutUIElements(withInferenceViewHeight height: CGFloat) {
+    pickFromGalleryButtonBottomSpace.constant =
+    height + Constants.pickFromGalleryButtonInset
+    view.layoutSubviews()
+  }
+
+  func redrawBoundingBoxesForCurrentDeviceOrientation() {
+    guard let imageSegmenterService = imageSegmenterService,
+          imageSegmenterService.runningMode == .image ||
+            self.playerViewController?.player?.timeControlStatus == .paused else {
+      return
+    }
+  }
+
+  deinit {
+    playerViewController?.player?.removeTimeObserver(self)
+  }
+}
+
+extension MediaLibraryViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+
+  func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+    picker.dismiss(animated: true)
+  }
+
+  private func playVideoAnDetect(asset: AVAsset) {
+    let videoDescription = getVideoFormatDescription(from: asset)
+    guard let formatDescription = videoDescription.description else { return }
+    if playerViewController == nil {
+      let playerViewController = AVPlayerViewController()
+      self.playerViewController = playerViewController
+    }
+
+    let playerItem = AVPlayerItem(asset: asset)
+
+    if let player = playerViewController?.player {
+      player.replaceCurrentItem(with: playerItem)
+    }
+    else {
+      playerViewController?.player = AVPlayer(playerItem: playerItem)
+    }
+
+    playerViewController?.showsPlaybackControls = false
+    addPlayerViewControllerAsChild()
+    guard let player = playerViewController?.player, let playerItem = player.currentItem else { return }
+    let timeRange = CMTimeRange(start: .zero, duration: asset.duration)
+    let videoComposition = AVMutableVideoComposition(asset: asset) { [weak self] request in
+      guard let self = self else { return }
+      backgroundQueue.async {
+        let sourceImage = request.sourceImage
+        let time = Float((request.compositionTime - timeRange.start).seconds)
+        let cgimage = self.render.getCGImmage(ciImage: sourceImage)
+        guard let resultBundle = self.imageSegmenterService?.segment(by: cgimage, orientation: .up, timeStamps: Int(time * 1000)) else {
```

Comment:
Like in the previous samples, please encapsulate all this code in the ImageSegmenterSevice method segment(
    videoAsset: AVAsset,
    durationInMilliseconds: Double,
    inferenceIntervalInMilliseconds: Double) async -> ResultBundle?